schism/compiler.ss

```diff
@@ -73,14 +73,13 @@
        (define (display x)
          (cond
           ((pair? x)
-           (begin (%display-raw-string "(")
+           (begin (%log-char #\()
                   (display (car x))
                   (%display-pair-tail (cdr x))))
-          ((null? x)
-           (%display-raw-string "()"))
+          ((null? x) (begin (%log-char #\() (%log-char #\))))
           ((symbol? x) (%display-raw-string (symbol->string x)))
-          ((boolean? x) (%display-raw-string (if x "#t" "#f")))
-          ((number? x) (begin (%display-leading-digits x) (%display-least-significant-digit x)))
+          ((boolean? x) (begin (%log-char #\#) (%log-char (if x #\t #\f))))
+          ((number? x) (begin (%display-leading-digits (/ x 10)) (%display-least-significant-digit x)))
```

Comment:
Ah I should mention that this change as well fixes a double-print issue for the least-significant digit.

Reply:
Thanks for fixing this!